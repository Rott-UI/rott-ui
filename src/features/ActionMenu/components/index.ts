export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
