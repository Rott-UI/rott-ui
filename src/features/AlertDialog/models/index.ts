export * from './alertDialogButtonProps'
export * from './alertDialogContextModel'
export * from './alertDialogModel'
