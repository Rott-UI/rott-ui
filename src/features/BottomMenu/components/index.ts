export * from './BottomMenu'
export * from './BottomMenuItem'
