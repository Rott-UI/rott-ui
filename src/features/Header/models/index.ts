export * from './HeaderIconProps'
export * from './HeaderProps'
