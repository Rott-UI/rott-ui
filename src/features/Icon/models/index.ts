export * from './iconProps'
export * from './iconType'
export * from './iconMode'
