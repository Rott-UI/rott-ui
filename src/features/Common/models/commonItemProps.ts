// React Native Imports
import {Animated} from 'react-native'

// Component Imports
import {CommonUiProps, Variant} from '../../../models'
import {IconProps, IconTypes} from '../..'
import {LabelProps} from '../../Label/models'

// Package Imports
import {Swipeable} from 'react-native-gesture-handler'
import {ReactNode} from 'react'

export interface CommonItemProps extends CommonUiProps {
  testID?: string
  index?: number
  width?: number
  height?: number
  backgroundColor?: Variant

  leftIcon?: IconTypes | IconProps
  rightIcon?: IconTypes | IconProps

  title: string | LabelProps | ReactNode
  subTitle?: string | LabelProps | ReactNode
  description?: string | LabelProps | ReactNode

  showSelected?: boolean
  selected?: boolean
  swipeable?: boolean
  favorite?: boolean

  renderRightActions?: (
    progressAnimatedValue: Animated.AnimatedInterpolation<string | number>,
    dragAnimatedValue: Animated.AnimatedInterpolation<string | number>,
    swipeable: Swipeable
  ) => React.ReactNode
  renderLeftActions?: (
    progressAnimatedValue: Animated.AnimatedInterpolation<string | number>,
    dragAnimatedValue: Animated.AnimatedInterpolation<string | number>,
    swipeable: Swipeable
  ) => React.ReactNode
  onPress?: (value?: any) => void | undefined
  value?: any
}
