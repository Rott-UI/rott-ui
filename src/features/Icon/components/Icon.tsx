/* eslint-disable indent */
// React Imports
import React, {FC} from 'react'

// Component Imports
import * as Icons from '../../../constants/Icons'
import {Item} from '../../../components/Item'

// Package Imports
import {colourFromVariant, display} from '../../../utils'
import {IconProps} from '../models'

export const Icon: FC<IconProps> = ({
  name,
  width = 16,
  height = 16,
  variant = 'white',
  strokeWidth = 1,
  mode = 'stroke',
  noStroke,
  color,
  ...props
}) => {
  const IconComponent = Icons[name]

  if (!IconComponent) return null

  return (
    <Item {...props}>
      <IconComponent
        fill={mode === 'fill' ? color ?? colourFromVariant(variant) : 'transparent'}
        stroke={
          mode === 'stroke'
            ? color ?? colourFromVariant(variant)
            : color ?? colourFromVariant(variant)
        }
        strokeWidth={
          mode === 'fill' && noStroke
            ? 0
            : mode === 'fill'
              ? 0.5
              : mode === 'stroke' && noStroke
                ? 0
                : strokeWidth
        }
        strokeLinecap='round'
        strokeLinejoin='round'
        width={display.px(width)}
        height={display.px(height)}
      />
    </Item>
  )
}
