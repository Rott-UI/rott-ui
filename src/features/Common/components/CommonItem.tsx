// React Imports
import {FC, isValidElement, memo} from 'react'

// Constants Imports

// Component Imports
import {Item} from '../../Item'
import {CommonItemProps} from '../models'
import {Pressable} from '../../Pressable'
import {IconProps} from '../../Icon/models'
import {Icon} from '../../Icon/components'
import {Label, LabelProps} from '../../Label'

// Util and Lib Imports
import {colorFromVariant} from '../../../utils'

// Package Imports
import Swipeable from 'react-native-gesture-handler/Swipeable'

/**
 *  Common Item Component
 * @param width - Item Genişliği
 * @param height - Item Yüksekliği
 * @param backgroundColor - Item Arka Plan Rengi varsayilan degeri `green-vogue`
 *
 * @param leftIcon - Solda renderlanacak icon, iconProps objesi alabilir, string olabilir veya react elementi alabilir
 * @param rightIcon - Sağda renderlanacak icon, iconProps objesi alabilir, string olabilir veya react elementi alabilir
 *
 * @param title - Başlık **Zorunlu**
 * @param subTitle - Alt Başlık
 * @param description - Açıklama
 *
 * @param showSelected - Seçili Gösterilsin mi
 * @param selected - Seçili mi
 * @param swipeable - Kaydirma Efekti
 * @param renderRightActions - Kaydirma Efektinde Sag Tarafta Renderlanacak Item Degerleri
 * @param renderLeftActions - Kaydirma Efektinde Sol Tarafta Renderlanacak Item Degerleri
 * @param onPress - Tıklandığında Alacağı Fonksiyon
 *
 * @returns Standart List Item Renderlar
 */
export const CommonItem: FC<CommonItemProps> = memo(
  ({
    testID,
    width,
    height,
    backgroundColor = 'grey-200',

    leftIcon,
    rightIcon,

    title,
    subTitle,
    description,

    swipeable,
    selected = false,
    showSelected,

    paddingVertical,
    paddingHorizontal,

    renderRightActions,
    renderLeftActions,
    onPress,
    value,
  }) => {
    const listItemContent = (
      <Pressable
        testID={testID}
        backgroundColor={colorFromVariant(backgroundColor)}
        onPress={() => onPress && onPress!(value)}
        paddingVertical={paddingVertical ?? 16}
        paddingHorizontal={paddingHorizontal ?? 24}
        width={width}
        height={height}>
        <Item row alignItemsCenter justifyContentCenter>
          {/* Left Icon / Element */}
          {leftIcon && (
            <Pressable
              flex={0}
              marginRight={16}
              {...(typeof leftIcon === 'object' ? {...leftIcon} : null)}>
              {isValidElement(leftIcon) && <>{leftIcon}</>}
              {!isValidElement(leftIcon) && (
                <Item width={24} height={24}>
                  <Icon
                    testID='left-icon-test-id'
                    width={24}
                    height={24}
                    variant={(leftIcon as IconProps)?.variant ?? 'grey-900'}
                    mode={(leftIcon as IconProps)?.mode ?? 'stroke'}
                    noStroke={
                      (leftIcon as IconProps)?.noStroke ?? (leftIcon as IconProps).mode === 'fill'
                    }
                    {...(typeof leftIcon === 'object' ? (leftIcon as IconProps) : {})}
                    name={(leftIcon as IconProps)?.name ?? leftIcon}
                  />
                </Item>
              )}
            </Pressable>
          )}
          {/* Left Icon / Element */}

          {/* Middle Section / Title, Subtitle, Description */}
          <Item flex={1}>
            {isValidElement(title) && <>{title}</>}
            {!isValidElement(title) && (
              <Label
                testID='title-test-id'
                variant={(title as LabelProps)?.variant ?? 'grey-900'}
                fontSize={(title as LabelProps)?.fontSize ?? 'lg'}
                fontFamily={(title as LabelProps)?.fontFamily ?? 'Markpro-Medium'}
                {...(typeof title === 'object' ? {...title} : null)}>
                {typeof title === 'string' ? title : (title as LabelProps)?.text}
              </Label>
            )}

            {subTitle && (
              <>
                {isValidElement(subTitle) && <>{subTitle}</>}
                {!isValidElement(subTitle) && (
                  <Label
                    testID='subtitle-test-id'
                    variant={(subTitle as LabelProps)?.variant ?? 'grey-900'}
                    fontSize={(subTitle as LabelProps)?.fontSize ?? 'md'}
                    fontFamily={(subTitle as LabelProps)?.fontFamily ?? 'Markpro-Medium'}
                    marginTop={8}
                    {...(typeof subTitle === 'object' ? {...subTitle} : null)}>
                    {typeof subTitle === 'string' ? subTitle : (subTitle as LabelProps)?.text}
                  </Label>
                )}
              </>
            )}

            {description && (
              <>
                {isValidElement(description) && <>{description}</>}
                {!isValidElement(description) && (
                  <Label
                    testID='description-test-id'
                    variant={(description as LabelProps)?.variant ?? 'grey-900'}
                    fontSize={(description as LabelProps)?.fontSize ?? 'md'}
                    fontFamily={(description as LabelProps)?.fontFamily ?? 'Markpro-Medium'}
                    marginTop={8}
                    {...(typeof description === 'object' ? {...description} : null)}>
                    {typeof description === 'string'
                      ? description
                      : (description as LabelProps)?.text}
                  </Label>
                )}
              </>
            )}
          </Item>
          {/* Middle Section / Title, Subtitle, Description */}

          {/* Right Icon / Element */}
          {!showSelected && rightIcon && (
            <>
              <Pressable
                flex={0}
                marginLeft={(rightIcon as IconProps)?.marginLeft ?? 16}
                {...(typeof rightIcon === 'object' ? {...rightIcon} : null)}>
                {!showSelected && isValidElement(rightIcon) && <>{rightIcon}</>}
                {!showSelected && !isValidElement(rightIcon) && (
                  <Item width={24} height={24}>
                    <Icon
                      testID='right-icon-test-id'
                      width={24}
                      height={24}
                      variant={(rightIcon as IconProps).variant ?? 'grey-900'}
                      mode={(rightIcon as IconProps).mode ?? 'stroke'}
                      noStroke={
                        (rightIcon as IconProps).noStroke ??
                        (rightIcon as IconProps).mode === 'fill'
                      }
                      {...(typeof rightIcon === 'object' ? (rightIcon as IconProps) : {})}
                      name={(rightIcon as IconProps).name ?? rightIcon}
                    />
                  </Item>
                )}
              </Pressable>
            </>
          )}

          {showSelected && (
            <Item
              width={24}
              height={24}
              marginLeft={8}
              borderRadius={24}
              borderWidth={2}
              borderColor={selected ? COLOURS.PRIMARY : COLOURS.GREY200}
              justifyContentCenter
              alignItemsCenter>
              {selected && (
                <Item
                  width={16}
                  height={16}
                  borderRadius={13}
                  backgroundColor={COLOURS.PRIMARY}
                  // eslint-disable-next-line react-native/no-inline-styles
                  style={{
                    shadowColor: COLOURS.NEUTRAL_BLUESOFT,
                    shadowOffset: {
                      width: 0,
                      height: 0,
                    },
                    shadowOpacity: 1,
                    shadowRadius: 1,
                    borderWidth: 2,
                    borderColor: COLOURS.NEUTRAL_BLUESOFT,
                  }}
                />
              )}
            </Item>
          )}
          {/* Right Icon / Element */}
        </Item>
      </Pressable>
    )

    return (
      <Item>
        {swipeable && (
          <Swipeable renderRightActions={renderRightActions} renderLeftActions={renderLeftActions}>
            {listItemContent}
          </Swipeable>
        )}

        {!swipeable && {...listItemContent}}
      </Item>
    )
  }
)
