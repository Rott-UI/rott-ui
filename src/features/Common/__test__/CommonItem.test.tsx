// React Imports
import React from 'react'

// Component Imports
import {Icon} from '../../Icon/components'
import {Label} from '../../Label/components'
import {CommonItemProps} from '../models'
import {Pressable} from '../../Pressable/components'
import {CommonItem} from '../components' // Import the CommonItem component

// Constants Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {formatMessage, render} from '../../../utils'

describe('List Item -> Custom Component', () => {
  const dummyData: CommonItemProps = {
    title: formatMessage('TEST.WITH.PARAM', {testText: 'TITLE'}),
    subTitle: formatMessage('TEST.WITH.PARAM', {testText: 'SUBTITLE'}),
    description: formatMessage('TEST.WITH.PARAM', {testText: 'DESCRIPTION'}),
    leftIcon: {
      name: 'STAR',
      width: 24,
      height: 24,
      color: COLOURS.PRIMARY,
    },
    rightIcon: {
      name: 'ADD',
      width: 24,
      height: 24,
      color: COLOURS.PRIMARY,
    },
  }

  const testId = {
    titleTestId: 'title-test-id',
    subTitleTestId: 'subtitle-test-id',
    descriptionTestId: 'description-test-id',
    leftIconTestId: 'left-icon-test-id',
    rightIconTestId: 'right-icon-test-id',
  }

  it('common item ilk renderlandığında snapshot ile eşleşmeli', () => {
    const commonItem = render(<CommonItem title={dummyData.title} />)

    expect(commonItem).toMatchSnapshot()
  })

  it('common item ilk renderlandığında title ekranda renderlanmali', () => {
    const {titleTestId} = testId
    const {title} = dummyData
    const {getByTestId} = render(<CommonItem title={title} />)

    const titleElement = getByTestId(titleTestId)

    expect(titleElement).toBeOnTheScreen()
  })

  it('common item ilk renderlandığında subTitle ekranda renderlanmali', () => {
    const {subTitleTestId} = testId
    const {title, subTitle} = dummyData
    const {getByTestId} = render(<CommonItem title={title} subTitle={subTitle} />)

    const subTitleElement = getByTestId(subTitleTestId)

    expect(subTitleElement).toBeOnTheScreen()
  })

  it('common item ilk renderlandığında description ekranda renderlanmali', () => {
    const {descriptionTestId} = testId
    const {title, subTitle, description} = dummyData
    const {getByTestId} = render(
      <CommonItem title={title} subTitle={subTitle} description={description} />
    )

    const descriptionElement = getByTestId(descriptionTestId)

    expect(descriptionElement).toBeOnTheScreen()
  })

  it('common item ilk renderlandığında leftIcon ekranda renderlanmali', () => {
    const {leftIconTestId} = testId
    const {title, subTitle, description, leftIcon} = dummyData
    const {queryByTestId} = render(
      <CommonItem title={title} subTitle={subTitle} description={description} leftIcon={leftIcon} />
    )

    const leftIconElement = queryByTestId(leftIconTestId)

    expect(leftIconElement).toBeOnTheScreen()
  })

  it('common item ilk renderlandiginda left iconu ekranda renderlanmamali', () => {
    const {leftIconTestId} = testId
    const {title, subTitle, description} = dummyData
    const {queryByTestId} = render(
      <CommonItem title={title} subTitle={subTitle} description={description} />
    )

    const leftIconElement = queryByTestId(leftIconTestId)

    expect(leftIconElement).toBeNull()
  })

  it('common item ilk renderlandığında rightIcon ekranda renderlanmali', () => {
    const {rightIconTestId} = testId
    const {title, subTitle, description, rightIcon} = dummyData
    const {getByTestId} = render(
      <CommonItem
        title={title}
        subTitle={subTitle}
        description={description}
        rightIcon={rightIcon}
      />
    )

    const rightIconElement = getByTestId(rightIconTestId)

    expect(rightIconElement).toBeOnTheScreen()
  })

  it('common item ilk renderlandiginda right iconu ekranda renderlanmamali', () => {
    const {rightIconTestId} = testId
    const {title, subTitle, description, leftIcon} = dummyData
    const {queryByTestId} = render(
      <CommonItem title={title} subTitle={subTitle} description={description} leftIcon={leftIcon} />
    )

    const rightIconElement = queryByTestId(rightIconTestId)

    expect(rightIconElement).toBeNull()
  })

  it('common item ilk renderlangidinga left ve right iconu ekranda renderlanmali', () => {
    const {rightIconTestId, leftIconTestId} = testId
    const {title, subTitle, description, rightIcon, leftIcon} = dummyData
    const {getByTestId} = render(
      <CommonItem
        title={title}
        subTitle={subTitle}
        description={description}
        rightIcon={rightIcon}
        leftIcon={leftIcon}
      />
    )

    const rightIconElement = getByTestId(rightIconTestId)
    const leftIconElement = getByTestId(leftIconTestId)

    expect(rightIconElement).toBeOnTheScreen()
    expect(leftIconElement).toBeOnTheScreen()
  })

  it('common item ilk renderlandiginda left ve right iconu ekranda renderlanmamali', () => {
    const {rightIconTestId, leftIconTestId} = testId
    const {title, subTitle, description} = dummyData
    const {queryByTestId} = render(
      <CommonItem title={title} subTitle={subTitle} description={description} />
    )

    const rightIconElement = queryByTestId(rightIconTestId)
    const leftIconElement = queryByTestId(leftIconTestId)

    expect(rightIconElement).toBeNull()
    expect(leftIconElement).toBeNull()
  })

  it('swipe ozelliginde sag eleman var ise ekranda renderlanmali', () => {
    const {title} = dummyData
    const rightElementTestId = 'right-element-test-id'
    const {getByTestId} = render(
      <CommonItem
        title={title}
        swipeable
        renderRightActions={() => (
          <Pressable
            alignItemsCenter
            justifyContentCenter
            backgroundColor={COLOURS.SECONDARY}
            testID={rightElementTestId}>
            <Icon name='STAR' width={24} height={24} />
            <Label>{formatMessage('LIST.ITEM.FAVORITE.ADD')}</Label>
          </Pressable>
        )}
      />
    )

    const rightElement = getByTestId(rightElementTestId)

    expect(rightElement).toBeOnTheScreen() // Right Element Ekranda Olmali
    expect(rightElement.children[0]).toHaveProp('name', 'STAR') // Icon Ekranda Olmali
    expect(rightElement).toHaveTextContent(formatMessage('LIST.ITEM.FAVORITE.ADD')) // Right Element Text Ekranda Olmali
  })

  it('swipe ozelliginde sol eleman var ise ekranda renderlanmali', () => {
    const {title} = dummyData
    const leftElementTestId = 'left-element-test-id'
    const {getByTestId} = render(
      <CommonItem
        title={title}
        swipeable
        renderLeftActions={() => (
          <Pressable
            alignItemsCenter
            justifyContentCenter
            backgroundColor={COLOURS.SECONDARY}
            testID={leftElementTestId}>
            <Icon name='STAR' width={10} height={10} />
            <Label>{formatMessage('LIST.ITEM.FAVORITE.ADD')}</Label>
          </Pressable>
        )}
      />
    )

    const leftElement = getByTestId(leftElementTestId)

    expect(leftElement).toBeOnTheScreen() // Right Element Ekranda Olmali
    expect(leftElement.children[0]).toHaveProp('name', 'STAR') // Icon Ekranda Olmali
    expect(leftElement).toHaveTextContent(formatMessage('LIST.ITEM.FAVORITE.ADD')) // Right Element Text Ekranda Olmali
  })
})
