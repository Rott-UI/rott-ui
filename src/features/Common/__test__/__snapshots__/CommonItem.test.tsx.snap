// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item -> Custom Component common item ilk renderlandığında snapshot ile eşleşmeli 1`] = `
<View
  style={
    {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "column",
      "flexGrow": undefined,
      "gap": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animated={false}
    backgroundColor="rgba(161, 173, 175, 1)"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    paddingHorizontal={24}
    paddingVertical={16}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": "rgba(161, 173, 175, 1)",
        "bottom": undefined,
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": 46,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": 31,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
  >
    <View
      alignItemsCenter={true}
      justifyContentCenter={true}
      style={
        {
          "alignItems": "center",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <View
        flex={1}
        style={
          {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
      >
        <Text
          style={
            {
              "backgroundColor": undefined,
              "bottom": undefined,
              "color": "rgba(34, 63, 70, 1)",
              "flexGrow": undefined,
              "fontFamily": "Markpro-Medium",
              "fontSize": 16,
              "fontWeight": undefined,
              "gap": undefined,
              "height": undefined,
              "left": undefined,
              "letterSpacing": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "maxHeight": undefined,
              "maxWidth": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "overflow": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "paddingVertical": undefined,
              "position": undefined,
              "right": undefined,
              "textAlign": "auto",
              "top": undefined,
              "width": undefined,
              "zIndex": undefined,
            }
          }
          testID="title-test-id"
        >
          Title
        </Text>
      </View>
    </View>
  </View>
</View>
`;
